docs/04.guides/04.cookbooks/29.Threads-Scope/page.md

```diff
@@ -0,0 +1,272 @@
+---
+title: Threads
+id: thread_usage
+---
+## Thread Scope ##
+
+This document explains how to use the threads. Threads mainly used for execute the code parallel.
+
+### Example 1 ###
+
+This below example shows waiting normal execution waiting for all code to resolve. Here it waits for 1000 mili seconds after the it complete the execution.
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	sleep(1000);
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+This below example shows thread execution. Code with in the thread tag run parallel to execution. Here execution doesn't wait for sleep.
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	thread {
+		sleep(1000);
+	}
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+Threads were independent form current running thread on lucee. Lucee doesn't wait for end of thread.
+
+Threads were mainly used to retrieve the data form DB, cfhttp, webservice it doesn't care about how much time it takes to execute.
+
+### Example 2 ###
+
+Here we can see about multiple threads in lucee. Lucee can also used to create multiple threads running parallel at that time.
+
+```lucee
+<cfscript>
+function napASecond(index) localmode=true {
+	thread  {
+		thread.start=now();
+		sleep(1000)
+		thread.end=now();
+	}
+}
+
+start=getTickCount();
+loop from=1 to=5 index="index" {
+	napASecond(index);
+}
+
+// show threads
+dump(var:cfthread,expand:false);
+// wait for all threads to finish
+thread action="join" name=cfthread.keyList();
+// show threads
+dump(var:cfthread,expand:false);
+
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+Above example shows currently five threads were running parallel, Thread action join wait for all threads to finish.
```

Comment:
***ToChange***
The above example shows five threads running in parallel. The _thread action= "join"_ line waits for all threads to finish.